client/src/modules/Course/pages/Student/CrossCheckSubmit/index.tsx

```diff
@@ -0,0 +1,218 @@
+              <CourseTaskSelect data={courseTasks} groupBy="deadline" onChange={handleTaskChange} />
+            ) : null}
+            {courseTasks.length === 0 ? <NoSubmissionAvailable courseAlias={props.course.alias} /> : null}
+            <SubmittedStatus
+              taskExists={taskExists}
+              solution={submittedSolution}
+              deadlinePassed={submitDeadlinePassed}
+            />
+            {submitAllowed && (
+              <Form.Item
+                name="url"
+                label="Solution URL"
+                rules={[
+                  {
+                    required: true,
+                    pattern: urlWithIpPattern,
+                    message: 'Please provide a valid link (must start with `http://` or `https://`)',
+                  },
+                ]}
+              >
+                <Input placeholder="link in the form of https://www.google.com" />
+              </Form.Item>
+            )}
+            {submitAllowed && newCrossCheck && (
+              <Form.Item name="review">
+                <CriteriaForm
+                  authorId={authorId ?? 0}
+                  onChange={handleReviewChange}
+                  criteria={criteria}
+                  comments={comments ?? []}
+                  reviewComments={newComments ?? []}
+                />
+              </Form.Item>
+            )}
+            {submitAllowed && newCrossCheck && <ScoreInput courseTask={task} />}
+            {submitAllowed && (
+              <Row style={{ marginTop: 16 }}>
+                <Col span={12}>
+                  <Button type="primary" htmlType="submit">
+                    Submit
+                  </Button>
+                </Col>
+                {submittedSolution && (
+                  <Col span={12} style={{ display: 'flex', justifyContent: 'flex-end' }}>
+                    <Button danger type="ghost" onClick={showModal}>
+                      Cancel Submit
+                    </Button>
+                    <Modal
+                      title="Cancel submission"
+                      visible={isModalVisible}
+                      onOk={handleCancellation}
+                      onCancel={cancelModal}
+                      okButtonProps={{ disabled: buttonDisabled }}
+                    >
+                      <Checkbox checked={!buttonDisabled} onChange={cancellationChange}>
+                        Being of sound mind and body, do hereby declare that I want to cancel my submission
+                      </Checkbox>
+                    </Modal>
+                  </Col>
+                )}
+              </Row>
+            )}
+          </Form>
+        </Col>
+      </Row>
+      <Row>
+        <CrossCheckComments comments={feedbackComments} />
+      </Row>
+    </PageLayout>
+  );
+}
+
+function calculateFinalScore(review: { percentage: number; criteriaId: string }[], criteria: CrossCheckCriteria[]) {
+  return review?.reduce((acc, r) => {
```

Comment:
Seems that `?` is redundant.

Reply:
not sure. typing might be broken here. added default parameter